File: backend/core-api/src/main/java/ai/govbiz/core/dto/sample/SampleItemRequest.java
package ai.govbiz.core.dto.sample;

import ai.govbiz.core.domain.sample.SampleCategory;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/** 샘플 항목의 공개 입력 DTO입니다. */
public record SampleItemRequest(
        @NotBlank @Size(max = 100) String name,
        SampleCategory category,
        @Size(max = 500) String note
) {
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/_sampleitem/controller/SampleItemRequest.kt
package ai.govbiz.core._sampleitem.controller

import ai.govbiz.core._sampleitem.domain.SampleCategory
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

/** 샘플 항목의 공개 입력 DTO입니다. */
data class SampleItemRequest(
    @field:NotBlank
    @field:Size(max = 100)
    val name: String?,
    val category: SampleCategory?,
    @field:Size(max = 500)
    val note: String?,
)
